Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -714,17 +719,18 @@ static inline RenderBlock* nearestNonAnonymousContainingBlockIncludingSelf(Rende
     return downcast<RenderBlock>(renderer);
 }
 
+RenderBlock* RenderObject::containingBlockForObjectInFlow() const
+{
+    auto* ancestor = parent();
+    while (ancestor && ((ancestor->isInline() && !ancestor->isReplacedOrAtomicInline()) || !ancestor->isRenderBlock()))
+        ancestor = ancestor->parent();
+    return downcast<RenderBlock>(ancestor);
+};
+
```

Comment:
any reason why this highly specific function is decoupled from containingBlockForPositionType? Are we expecting more clients of this function here?

Reply:
Just the one that I added in this patch, but I can go through `containingBlockForPositionType` instead.